bin/check_sm_processing_job.py

```diff
@@ -0,0 +1,138 @@
+        region_name = sagemaker_client.meta.region_name
+        cw_endpoint = f"https://{region_name}.console.aws.amazon.com/cloudwatch/home"
+        cloudwatch_url = (
+            f"{cw_endpoint}?region={region_name}#logsV2:log-groups/log-group/{log_group}/log-events/{log_stream}"
+        )
+
+        return status, duration, creation_time.strftime("%Y-%m-%d %H:%M:%S %Z"), cloudwatch_url
+    except sagemaker_client.exceptions.ResourceNotFound:
+        return "Processing job not found."
+    except Exception as e:
+        raise Exception(f"An error occurred: {str(e)}")
+
+
+def duration_in_human_readable(start_time: datetime, end_time: datetime) -> str:
+    """
+    Calculate the duration between start and end times in a human-readable format.
+
+    :param start_time: Start time.
+    :param end_time: End time.
+
+    :returns: Duration in a human-readable format (e.g., '1 day, 2 hours, 30 minutes').
+    """
+    duration = end_time - start_time
+    days = duration.days
+    hours, remainder = divmod(duration.seconds, 3600)
+    minutes, seconds = divmod(remainder, 60)
+
+    if days > 0:
+        return f"{days} days, {hours} hours, {minutes} minutes"
+    elif hours > 0:
+        return f"{hours} hours, {minutes} minutes"
+    elif minutes > 0:
+        return f"{minutes} minutes"
+    else:
+        return "Less than a minute"
+
+
+def get_recent_processing_jobs(sagemaker_client: ServiceResource, max_results: int) -> Union[List[Dict], str]:
+    """
+    Get the most recent SageMaker processing jobs.
+
+    :param sagemaker_client: The SageMaker client.
+    :param max_results: The maximum number of recent jobs to retrieve.
+
+    :returns: A list of processing job summaries or an error message.
+    """
+    try:
+        response = sagemaker_client.list_processing_jobs(
+            SortBy="CreationTime", SortOrder="Descending", MaxResults=max_results
+        )
+        job_summaries = response["ProcessingJobSummaries"]
+        return job_summaries
+    except Exception as e:
+        raise Exception(f"An error occurred: {str(e)}")
+
+
+def print_job_status_table(sagemaker_client: ServiceResource, jobs: List[Dict], region: str) -> None:
+    """
+    Print the job statuses in a formatted table.
+
+    :param sagemaker_client: The SageMaker client.
+    :param jobs: A list of processing job summaries.
+    :param region: The AWS region where the SageMaker jobs are running.
+
+    :returns: None
+    """
+    print(f"{'Job Name':<60} {'Status':<15} {'Region':<15} {'Duration':<20} {'Creation Time':<30}")
+    print("=" * 150)
+
+    # Print the job details
+    for job_summary in jobs:
+        job_name = job_summary["ProcessingJobName"]
+        status, duration, creation_time, cloudwatch_url = get_processing_job(sagemaker_client, job_name)
+        print(f"{job_name:<60} {status:<15} {region:<15} {duration:<20} {creation_time:<30}")
+        print(f"\tCloudwatch URL: {cloudwatch_url}")
```

Comment:
Same as above - may want to switch to use the logger instead of print. That would also give the ability for a --verbose argument tied to the log level for more detailed output